### PR TITLE
add 2023-05-25 URTI progress report

### PR DESCRIPTION
**Pull request by mossmann**
phase one completion

### CODE REVIEW

File: project-reports/2023-05-25-urti-progress-report/README.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+5. Host software development to enable use of the mainboard as a USB peripheral
+6. Final mainboard prototype design
+7. User interface board hardware design
+8. Handheld user interface firmware development
+
+In phase one, we developed a preliminary architecture, a preliminary mainboard block diagram, and evaluated components and sub-circuits in preparation for the initial mainboard design.
+
+## Architecture
+
+We plan to build URTI as a system comprised of two printed circuit board assemblies: a mainboard, and a user interface board. The mainboard will include the radio section and a USB interface and will be capable of serving as a Software-Defined Radio peripheral with or without the user interface board attached. The user interface board will include tactile, controls, display, application processor, and battery, allowing handheld use of the complete system.
```
Comment (epenelope) [inline project-reports/2023-05-25-urti-progress-report/README.md]: remove the comma after mainboard
also, it's not clear what the adjective tactile is describing in the last sentence. If the adjective tactile is meant to describe the controls, we should remove the comma after tactile.

Comment (epenelope) [reply project-reports/2023-05-25-urti-progress-report/README.md]: The second sentence is a little bit hard to follow. I suggest splitting it into two sentences:
"The mainboard will include the radio section and a USB interface. It will be capable of serving as..."

File: project-reports/2023-05-25-urti-progress-report/README.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+The Analog-to-Digital Converters (ADCs) and Digital-to-Analog Converters (DACs) will be provided by MAX5865, a dual-ADC/dual-DAC component. MAX5865 is a higher speed grade of MAX5864, a component used on HackRF One. Because of our familiarity with MAX5864, we did not need to evaluate MAX5865 further.
+
+The ADCs will sample quadrature analog baseband signals for RX. The DACs will produce quadrature analog baseband signals for TX.
+
+A potential drawback of MAX5865 is the limited dynamic range of its 8-bit ADCs and 10-bit DACs. If we decide to upgrade to a part with more dynamic range in the future, [AFE7222](https://www.ti.com/product/AFE7222) is a likely candidate.
+
+### Quadrature Transmitter
+
+We will use [MAX2831](https://www.analog.com/en/products/max2831.html) or [MAX2830](https://www.analog.com/en/products/max2830.html) to convert quadrature analog baseband signals to an Intermediate Frequency (IF) between 2.4 GHz and 2.5 GHz for transmission. MAX2831 and MAX2830 are both very similar to MAX2837 used on HackRF One, but we chose to test the components individually.
```
Comment (epenelope) [inline project-reports/2023-05-25-urti-progress-report/README.md]: I think you could explain why we selected those two components for evaluation. Reading this, it sounds like you are saying MAX2837 was also a candidate for URTI. We know why it wasn't a candidate, but the audience of this report might not have that context.

And why did we choose to test them individually? Is it because they were dissimilar enough that we couldn't rely on our familiarity with MAX2837? 

Finally, it's not clear what the word 'individually' means in the last sentence. Does it mean we tested each of those two components separately from MAX2837?

File: project-reports/2023-05-25-urti-progress-report/README.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+*testing MAX2830 with MAX2830EVKIT controlled by a GreatFET One*
+
+</figcaption>
+
+### Quadrature Receiver
+
+We will use [MAX2120](https://www.analog.com/en/products/max2120.html) to convert an IF between 925 MHz and 2175 MHz to quadrature analog baseband. MAX2120 is somewhat similar to MAX2837 used on HackRF One, but it is a receiver, not a transceiver. It includes a quadrature downconverter, a frequency synthesizer, baseband filters, and gain.
+
+We used [MAX2223EVKIT](https://www.analog.com/en/design-center/evaluation-hardware-and-software/evaluation-boards-kits/max2223evkit.html), an off-the-shelf evaluation board, to test [MAX2223](https://www.analog.com/en/products/max2223.html) and MAX2120. We controlled the device with a GreatFET One and connected the RF ports to an RF signal generator and spectrum analyzer. MAX2223 is similar but lacks baseband filters. We tested MAX2223 first and then replaced the MAX2223 with MAX2120 on the MAX2223EVKIT to test MAX2120.
```
Comment (epenelope) [inline project-reports/2023-05-25-urti-progress-report/README.md]: The sentence "MAX2223 is similar but lacks baseband filters" doesn't make sense to me in this paragraph. What is it similar to?

File: project-reports/2023-05-25-urti-progress-report/README.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+The MAX2120's external clock input functioned with input all the way up to 60 MHz. Although the data sheet appeared to indicate that it should only work up to 8 MHz, we later reinterpreted the data sheet to mean that it is guaranteed to work up to 30 MHz. (There are several other errors and omissions in the data sheet. For example, the register addresses are all off by one. Testing with the very similar MAX2223 first and reading its data sheet helped us understand the MAX2120 quite a lot.)
+
+The MAX2120 tuning step size must be between 1 and 2 MHz. It must be on the upper end of that range if using a reference clock frequency higher than 30 MHz.
+
+RF gain control worked, but it requires an analog voltage generated by a DAC. There is an additional 15 dB of digitally controlled baseband gain, but it seemed to amplify the noise floor just as much as the signal. We may pick a fixed setting to match the ADC and then adjust only the RF gain.
+
+We tested MAX2223 because it was the component initially installed on the evaluation board, but we plan to use MAX2120 because of its analog baseband filter and low cost.
```
Comment (epenelope) [inline project-reports/2023-05-25-urti-progress-report/README.md]: This seems out of place; I think this information belongs in line 87, where you first mention the component we chose.

File: project-reports/2023-05-25-urti-progress-report/README.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+</figcaption>
+
+### Quadrature Receiver
+
+We will use [MAX2120](https://www.analog.com/en/products/max2120.html) to convert an IF between 925 MHz and 2175 MHz to quadrature analog baseband. MAX2120 is somewhat similar to MAX2837 used on HackRF One, but it is a receiver, not a transceiver. It includes a quadrature downconverter, a frequency synthesizer, baseband filters, and gain.
+
+We used [MAX2223EVKIT](https://www.analog.com/en/design-center/evaluation-hardware-and-software/evaluation-boards-kits/max2223evkit.html), an off-the-shelf evaluation board, to test [MAX2223](https://www.analog.com/en/products/max2223.html) and MAX2120. We controlled the device with a GreatFET One and connected the RF ports to an RF signal generator and spectrum analyzer. MAX2223 is similar but lacks baseband filters. We tested MAX2223 first and then replaced the MAX2223 with MAX2120 on the MAX2223EVKIT to test MAX2120.
+
+Both components worked as expected. We observed that they are capable of tuning well outside their advertised ranges. MAX2223 operated from 513 MHz to 2653 MHz while MAX2120 operated from 499 MHz to 2442 MHz.
```
Comment (epenelope) [inline project-reports/2023-05-25-urti-progress-report/README.md]: The statement "both components worked as expected" seems to conflict with a statement in the following paragraph, "there are several other errors and omissions in the data sheet". If they worked as expected, what did we base our expectations on if not the datasheet?

File: project-reports/2023-05-25-urti-progress-report/README.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+</figcaption>
+
+### Mixers
+
+The transmit and receive paths will each use [RFFC5072](https://www.qorvo.com/products/p/RFFC5072) for conversion between IF and RF. RFFC5072 is a mixer with integrated frequency synthesizer capable of operation over a wide frequency range. Because we are familiar with the part as used on HackRF One, we did not need to evaluate it further.
+
+### SMA Connector and Trace Width
+
+Because of our need to design many small PCBs for testing RF components and sub-circuits throughout the project, it was important to establish a baseline test PCB design featuring RF traces with 50 ohm characteristic impedance and low-loss RF connectors at a reasonable cost. [Initial tests](../../rf-testing/2022-04-06-sma-and-trace-width-test) of PCBs manufactured by [JLCPCB](https://jlcpcb.com/) and [OSH Park](https://oshpark.com/) were moderately successful. [Further refinements](../../rf-testing/2022-04-24-sma-and-trace-width-test) helped us arrive upon a solution that we have used as a basis for several subsequent test boards.
```
Comment (epenelope) [inline project-reports/2023-05-25-urti-progress-report/README.md]: Replace the first sentence with: "Establishing a baseline test PCB design featuring RF traces with 50 ohm characteristic impedance and low-loss RF connectors at a reasonable cost was important because we will need to design many small PCBs for testing RF components and sub-circuits throughout the project."

Comment (epenelope) [inline project-reports/2023-05-25-urti-progress-report/README.md]: Replace "arrive upon a solution" with "arrive at a solution"

File: project-reports/2023-05-25-urti-progress-report/README.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+### Clock Generation
+
+While testing clock generators for a recent design change to HackRF One, we determined that the phase noise of CDEL925 and 5L1503 are too high for use in URTI. Si5351 (used in HackRF One) has lower phase noise and would be acceptable, but it may not be necessary to introduce even that much phase noise.
+
+HackRF One supports arbitrary sample rates by producing the sample clock with the very flexible Si5351. In contrast, URTI will be able to perform resampling in its FPGA, making it possible to use a fixed sample rate for the ADC and DAC.
+
+We plan to generate a single 40 MHz clock signal that is used as the ADC/DAC sample clock and also as a reference clock for each of the RF synthesizers. The mixers (RFFC5072) and quadrature transmitters (MAX2830 and MAX2831) support a 40 MHz reference clock directly. The quadrature receiver (MAX2120) can accept a 20 MHz reference clock which we will generate by dividing the sample clock by two with a flip-flop.
```
Comment (epenelope) [inline project-reports/2023-05-25-urti-progress-report/README.md]: Change the first sentence to "We plan to generate a single 40 MHz clock signal that is used as the ADC/DAC sample clock and as a reference clock for each RF synthesizer. "

File: project-reports/2023-05-25-urti-progress-report/README.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+In the initial mainboard prototype we plan to use a single 40 MHz temperature-compensated crystal oscillator (TCXO). In the final mainboard design we will likely use a voltage-controlled crystal oscillator (VCXO) capable of being disciplined by an external clock input.
```
Comment (epenelope) [inline project-reports/2023-05-25-urti-progress-report/README.md]: Change the first sentence to: "We plan to use a single 40 MHz temperature-compensated crystal oscillator (TCXO) in the initial mainboard prototype."

Add a comma after the word 'design' in the second sentence

File: project-reports/2023-05-25-urti-progress-report/README.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+
+### RF Limiter
+
+Although a limiter does not appear in the preliminary block diagram, we think that we will want to include one or more to protect RF components in the final mainboard design.
+
+Modern RF ICs of the sort we expect to use in URTI and related designs such as [Amalthea](https://github.com/greatscottgadgets/amalthea-hardware) often specify a maximum input power of 10 dBm to 20 dBm. Unfortunately, off-the-shelf RF limiter components and most [traditional designs](https://www.skyworksinc.com/-/media/SkyWorks/Documents/Products/1-100/200480C.pdf) do not begin to have an affect until the input power reaches at least 10 dBm, often much higher. The very wide operating frequency range and low cost requirements of URTI make this problem even more difficult to solve. While investigating this problem, we discovered solutions proposed by Chin-Leong Lim:
```
Comment (epenelope) [inline project-reports/2023-05-25-urti-progress-report/README.md]: Change 'affect' to 'effect'

Comment (epenelope) [reply project-reports/2023-05-25-urti-progress-report/README.md]: Change "The very wide operating frequency range and low cost requirements of URTI make this problem even more difficult to solve." to "URTI's wide operating frequency range and low-cost requirements make this problem even more challenging to solve."

File: project-reports/2023-05-25-urti-progress-report/README.md [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+* [Low Loss Configuration for Integrated PIN-Schottky Limiters](https://www.microwavejournal.com/articles/22741-low-loss-configuration-for-integrated-pin-schottky-limiters)
+
+Lim's key insight is that a Schottky diode anti-parallel to a PIN diode can be used to significantly reduce the limiting threshold of the PIN diode. Broadcom/Avago produced [ASML-5829](https://www.digikey.ca/en/articles/how-and-why-to-use-pin-diodes-for-rf-switching) and other PIN-Schottky limiters in a single package, but they are now obsolete. A few other PIN-Schottky limiters such as [SKY16602-632LF](https://www.skyworksinc.com/en/Products/Diodes/SKY16602-632LF) are available, but they are expensive for use on URTI.
+
+Another trick used by Lim is the use of the parasitic capacitance and inductance of the limiter diodes as elements in a low-pass filter (which can have a high cutoff frequency when using small diodes). This improves the impedance match in the passband, reducing loss. The earliest reference we found to this technique for RF limiters was in HP/Agilent Application Note 1050, *Low Cost Surface Mount Power Limiters*.
```
Comment (epenelope) [inline project-reports/2023-05-25-urti-progress-report/README.md]: Change "used by Lim is the use of" to "Lim uses is applying"

Comment (epenelope) [reply project-reports/2023-05-25-urti-progress-report/README.md]: To clarify the antecedent, add a word: "This _technique_ improves the impedance match in the passband..."

File: project-reports/2023-05-25-urti-progress-report/README.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+
+We [tested](../../rf-testing/2022-04-24-limiter-test/) a few PIN-Schottky solutions using discrete diodes and found that the technique is likely suitable for URTI.
+
+## Progress on Overlapping Phases
+
+During phase one we additionally made progress on phases four and eight. Although completion of the later phases is still a long way off, work toward those software development goals in parallel with hardware design tasks will speed completion of software after hardware is available. This approach has allowed us to involve more team members in the project.
```
Comment (epenelope) [inline project-reports/2023-05-25-urti-progress-report/README.md]: Add a comma after 'phase one'

Comment (epenelope) [reply project-reports/2023-05-25-urti-progress-report/README.md]: Change 'work' to 'working'

File: project-reports/2023-05-25-urti-progress-report/README.md [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+
+Separate from URTI but partly in an effort to provide a platform for URTI developers, we have designed and assembled [Amalthea](https://github.com/greatscottgadgets/amalthea-hardware), a Software-Defined Radio add-on for Cynthion. Amalthea provides both transmit and receive capabilities with the same Cynthion-based FPGA back-end as planned for URTI, using our [LUNA](https://github.com/greatscottgadgets/luna) gateware framework for USB. This platform has allowed us to begin development of a [Digital Signal Processing (DSP) library](https://github.com/mndza/dsp_sandbox) for [Amaranth HDL](https://github.com/amaranth-lang/amaranth) and other gateware that will be essential for URTI.
```
Comment (epenelope) [inline project-reports/2023-05-25-urti-progress-report/README.md]: Replace 'in an effort to' with just 'to'

**Review by epenelope — CHANGES_REQUESTED** [review]
Great start! I've commented in line with specific suggestions for refining edits.

**Review by epenelope — CHANGES_REQUESTED** [review]
Great start! I've commented in line with specific suggestions for refining edits.

File: project-reports/2023-05-25-urti-progress-report/README.md (hunk 10 above)
Comment (mossmann) [reply project-reports/2023-05-25-urti-progress-report/README.md]: I'm replacing "Another trick used by Lim is the use of" with "Lim also suggests using".

**Comment by mossmann** [comment]
Thank you for the great edits, @epenelope! I've addressed all your suggestions, added a short section on DPDT RF switch testing, and updated the report date.

**Review by epenelope — APPROVED** [review]
This looks great; thanks, @mossmann !
